services/src/datasets/external/natur40.rs

```diff
@@ -0,0 +1,290 @@
+        _dataset: &geoengine_datatypes::dataset::DatasetId,
+    ) -> crate::error::Result<crate::datasets::storage::Dataset> {
+        Err(error::Error::NotYetImplemented)
+    }
+}
+
+impl Natur40DataProvider {
+    fn auth(&self) -> [String; 2] {
+        [
+            format!("UserPwd={}:{}", self.user, self.password),
+            "HttpAuth=BASIC".to_owned(),
+        ]
+    }
+
+    async fn load_dataset(&self, db_url: String) -> Result<gdal::Dataset> {
+        let auth = self.auth();
+        tokio::task::spawn_blocking(move || {
+            let dataset = gdal_open_dataset_ex(
+                Path::new(&db_url),
+                DatasetOptions {
+                    open_options: Some(&[&auth[0], &auth[1]]),
+                    ..DatasetOptions::default()
+                },
+            )?;
+            Ok(dataset)
+        })
+        .await
+        .context(error::TokioJoin)?
+    }
+
+    async fn load_raster_dbs(&self) -> Result<RasterDbs> {
+        Client::new()
+            .get(format!("{}/rasterdbs.json", self.base_url))
+            .basic_auth(&self.user, Some(&self.password))
+            .send()
+            .await?
+            .json()
+            .await
+            .context(error::Reqwest)
+    }
+}
+
+#[async_trait]
+impl MetaDataProvider<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for Natur40DataProvider
+{
+    async fn meta_data(
+        &self,
+        dataset: &DatasetId,
+    ) -> Result<
+        Box<dyn MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>>,
+        geoengine_operators::error::Error,
+    > {
+        let dataset = dataset
+            .external()
+            .ok_or(geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(Error::InvalidExternalDatasetId { provider: self.id }),
+            })?;
+        let split: Vec<_> = dataset.dataset_id.split(':').collect();
+
+        let (db_name, band_index) = match *split.as_slice() {
+            [db, band_index] => {
+                if let Ok(band_index) = band_index.parse::<isize>() {
+                    (db, band_index)
+                } else {
+                    return Err(geoengine_operators::error::Error::LoadingInfo {
+                        source: Box::new(Error::InvalidExternalDatasetId { provider: self.id }),
+                    });
+                }
+            }
+            _ => {
+                return Err(geoengine_operators::error::Error::LoadingInfo {
+                    source: Box::new(Error::InvalidExternalDatasetId { provider: self.id }),
+                })
+            }
+        };
+
+        let db_url = RasterDb::url_from_name(&self.base_url, db_name);
+        let dataset = self.load_dataset(db_url.clone()).await.map_err(|e| {
+            geoengine_operators::error::Error::LoadingInfo {
+                source: Box::new(e),
+            }
+        })?;
+
+        Ok(Box::new(GdalMetaDataStatic {
+            time: None,
```

Comment:
Every dataset is static from this type of db?

Reply:
I think yes

Reply:
Maybe collect this (and the colorization) in a list of open topics to discuss for the Nature 4.0 guys?

Reply:
I added a checklist to the story card